src/main/java/ch/wisv/areafiftylan/products/service/OrderServiceImpl.java

```diff
@@ -168,16 +185,21 @@ public Order updateOrderStatus(String orderReference) {
     }
 
     private void validateTicketsIfPaid(Order order) {
+        if (order.getUser() == null) {
+            throw new UnassignedOrderException(order.getId());
+        }
+
         if (order.getStatus().equals(OrderStatus.PAID)) {
             for (Ticket ticket : order.getTickets()) {
+                ticketService.assignTicketToUser(ticket.getId(), order.getUser().getUsername());
```

Comment:
I would perform this action when assigning the order to the user. Since that is the point from where on the order, and tickets are assigned to a user. From that point on the user can start adding tickets under his own account.